cmake/compiler_utils.cmake

```diff
@@ -56,4 +56,31 @@ function(get_muparserx_source_code)
                 Please download MuparserX library from https://github.com/beltoforion/muparserx.git \
                 and checkout latest stable release")
     endif()
-endfunction()
\ No newline at end of file
+endfunction()
+
+function(check_compiler_cpp11_abi)
+    # Consider also if -D_GLIBCXX_USE_CXX11_ABI has been passed as flag
+    string(REGEX MATCH "-D_GLIBCXX_USE_CXX11_ABI=[(A-z)|(a-z)|(0-9)]+" CUSTOM_PREP_FLAGS ${CMAKE_CXX_FLAGS})
+    # Preprocessor run to check if CXX11_ABI is set
+    execute_process(COMMAND echo "#include <string>" COMMAND ${CMAKE_CXX_COMPILER} ${CUSTOM_PREP_FLAGS} -x c++ -E -dM -  COMMAND fgrep _GLIBCXX_USE_CXX11_ABI OUTPUT_VARIABLE C++11_ABI_OUT OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string(REGEX REPLACE "#define _GLIBCXX_USE_CXX11_ABI " "" C++11_ABI "${C++11_ABI_OUT}")
+    set(C++11_ABI ${C++11_ABI} PARENT_SCOPE)
```

Comment:
I'm surprised that cmake allows '+' as a variable name...

Reply:
Actually, I've seen problems with '+' in names so I'll change them.